nomt/src/seglog/mod.rs

```diff
@@ -1323,6 +1323,34 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn test_prune_front_segment_choice() -> Result<()> {
+        let h = TestHarness::new()?;
+        h.new_segment(1)?
+            .write_record(1, &[1u8; 10])?
+            .write_record(2, &[2u8; 10])?
+            .write_record(3, &[3u8; 10])?
+            .write()?;
+
+        h.new_segment(2)?
+            .write_record(4, &[4u8; 10])?
+            .write_record(5, &[5u8; 10])?
+            .write_record(6, &[6u8; 10])?
+            .write()?;
+
+        h.new_segment(3)?
+            .write_record(7, &[4u8; 10])?
+            .write_record(8, &[5u8; 10])?
+            .write_record(9, &[6u8; 10])?
+            .write()?;
+
+        let (mut log, _) = h.open_log(default_max_segment_size(), 1, 9)?;
+        assert_eq!(log.live_range(), (RecordId::from(1), RecordId::from(9)));
+        log.prune_front(5.into())?;
```

Comment:
with the old logic, this was choosing segment 1 (3 <= 5) and not segment 2 (6 > 5). Then failing because it couldn't find record 5 in segment 1.